### PR TITLE
chore(sweeps): reorder validation checks and improve messaging

### PR DESCRIPTION
**Pull request by gtarpenning**
Removes the scheduler job experimental warning, adds entity/project field handling from sweep config, and other misc messaging improvements.

https://wandb.atlassian.net/browse/WB-14428

### CODE REVIEW

File: wandb/cli/cli.py [hunk 1]
```diff
@@ -939,35 +939,42 @@ def launch_sweep(
         ctx.invoke(login, no_offline=True)
         api = _get_cling_api(reset=True)
 
-    entity = entity or env.get("WANDB_ENTITY") or api.settings("entity")
+    # if not sweep_config XOR resume_id
+    if not (config or resume_id):
+        wandb.termerror("'config' and/or 'resume_id' required")
+        return
+
+    parsed_user_config = sweep_utils.load_launch_sweep_config(config)
```
Comment (gtarpenning) [inline wandb/cli/cli.py:947]: parse config before entity/project setting, to include `project` and `entity` fields set in the sweep config.

File: wandb/cli/cli.py [hunk 2]
```diff
@@ -1033,6 +1040,14 @@ def launch_sweep(
         )
         return
 
+    # validate training job existence
+    if not sweep_utils.check_job_exists(PublicApi(), sweep_config.get("job")):
+        return False
+
+    # validate scheduler job existence, if present
+    if not sweep_utils.check_job_exists(PublicApi(), scheduler_job):
+        return False
```
Comment (gtarpenning) [inline wandb/cli/cli.py:1049]: this can happen before entrypoint creation

File: wandb/cli/cli.py [hunk 3]
```diff
@@ -1044,13 +1059,10 @@ def launch_sweep(
...
 
-    # validate training job existence
-    if not sweep_utils.check_job_exists(PublicApi(), sweep_config.get("job")):
-        return False
-
-    # validate scheduler job existence, if present
-    if not sweep_utils.check_job_exists(PublicApi(), scheduler_job):
-        return False
+    # set name of scheduler
+    name = scheduler_args.get("name") or "scheduler.WANDB_SWEEP_ID"
```
Comment (gtarpenning) [inline wandb/cli/cli.py:1063]: use the format the frontend uses, and pass it into the run setting object for consumption by the sweep scheduler.

File: wandb/sdk/launch/sweeps/scheduler.py [hunk 4]
```diff
@@ -246,9 +246,9 @@ def available_workers(self) -> Dict[int, _Worker]:
 
     def _init_wandb_run(self) -> SdkRun:
         """Controls resume or init logic for a scheduler wandb run."""
-        _type = self._kwargs.get("sweep_type", "sweep")
+        name = self._settings_config.get("name") or f"Scheduler.{self._sweep_id}"
```
Comment (gtarpenning) [inline wandb/sdk/launch/sweeps/scheduler.py:249]: load name from settings instead of overwritting

File: wandb/sdk/launch/sweeps/launch_sweep.py [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+        entity (str): The entity to launch the sweep under
+        project (str): The project to launch the sweep under
+        queue (str): The queue name to launch the sweep to
+        parsed_user_config (Optional[Dict[str, Any]]): a dictionary of sweep and launch config parameters
+        resume_id (Optional[str]): The id of the sweep to resume
+
+    Returns:
+        str: The sweep id of the launched sweep
+
+    Example:
```
Comment (nickpenaranda) [inline wandb/sdk/launch/sweeps/launch_sweep.py:38]: nit: Finish out the example w/ an invocation.

File: wandb/sdk/launch/sweeps/launch_sweep.py [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+    get_previous_args,
+    handle_sweep_config_violations,
+)
+
+
+def launch_sweep(
+    entity: str,
+    project: str,
+    queue: str,
+    parsed_user_config: Optional[Dict[str, Any]] = None,
```
Comment (nickpenaranda) [inline wandb/sdk/launch/sweeps/launch_sweep.py:23]: seems unnecessarily verbose. this is public api, do we ever expect the user to have their config in serialized form?  IMO just call this `config` or `config_overrides`?

File: wandb/sdk/launch/sweeps/launch_sweep.py [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+def _launch_sweep(
+    api: InternalApi,
+    entity: str,
+    project: str,
+    queue: str,
+    parsed_user_config: Optional[Dict[str, Any]] = None,
+    resume_id: Optional[str] = None,
+) -> Optional[str]:
+    # if not sweep_config XOR resume_id
+    if not (parsed_user_config or resume_id):
```
Comment (nickpenaranda) [inline wandb/sdk/launch/sweeps/launch_sweep.py:85]: What's the comment about? do we really mean XOR here? XOR is the same truth table as `a != b`, given a and b are booleans.

Comment (nickpenaranda) [reply wandb/sdk/launch/sweeps/launch_sweep.py:85]: oh but it's `NOT (a XOR b)` ? that's just `a == b`, again assuming a and b are bools :). A bit weird until you orient to the truth table perspective

File: wandb/sdk/launch/sweeps/launch_sweep.py [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+            "launch": {
+                "resource_args": {
+                    "local-container": {
+                        "env": ["WANDB_BASE_URL", "WANDB_SILENT"]
+                    }
+                }
+            }
+        }
+    """
+    api = InternalApi()
```
Comment (nickpenaranda) [inline wandb/sdk/launch/sweeps/launch_sweep.py:64]: it gets really hard to read from here on down.  are there significant changes in the code or is this just a straight extraction?

**Review by nickpenaranda — COMMENTED** [review]
small nits and a Q

File: wandb/sdk/launch/sweeps/launch_sweep.py (hunk 8 above)
Comment (gtarpenning) [reply wandb/sdk/launch/sweeps/launch_sweep.py:64]: Functionally an extraction, with slight re-ordering to improve logical flow. I'll put some comments in there.

File: wandb/sdk/launch/sweeps/launch_sweep.py (hunk 7 above)
Comment (gtarpenning) [reply wandb/sdk/launch/sweeps/launch_sweep.py:85]: Yeah, that comment is extra confusing.

File: wandb/cli/cli.py [hunk 9]
```diff
@@ -1500,50 +1422,6 @@ def agent(ctx, project, entity, count, sweep_id):
...
-                _args = int(_args)
-            kwargs[_key] = _args
-    try:
-        sweep_type = kwargs.get("sweep_type", "wandb")
-        _scheduler = load_scheduler(scheduler_type=sweep_type)(
-            api,
-            sweep_id=sweep_id,
-            **kwargs,
-        )
-        _scheduler.start()
```
Comment (gtarpenning) [inline wandb/cli/cli.py:1541]: remove arbitrary kwarg handling

**Pull request closed without merging**